File: frontend/crud-application/src/app/service/Valuation.ts
import { Child } from './Child';

export class Valuation {
  id: number;
  values: string[];
  childId: number;
  child: Child;
  createdAt: string;

  constructor(id: number, values: string[], childId: number, child: Child, createdAt: string) {
    this.id = id;
    this.values = values;
    this.childId = childId;
    this.child = child;
    this.createdAt = createdAt;
  }

  getId(): number {
    return this.id;
  }

  setId(id: number): void {
    this.id = id;
  }

  getValues(): string[] {
    return this.values;
  }

  setValues(values: string[]): void {
    this.values = values;
  }

  getChildId(): number {
    return this.childId;
  }

  setChildId(childId: number): void {
    this.childId = childId;
  }

  getChild(): Child {
    return this.child;
  }

  setChild(child: Child): void {
    this.child = child;
  }

  getCreatedAt(): string {
    return this.createdAt;
  }

  setCreatedAt(createdAt: string): void {
    this.createdAt = createdAt;
  }

  toString(): string {
    return `AriaValues{id=${this.id}, values=${this.values}, childId=${this.childId}, child=${this.child}, createdAt=${this.createdAt}}`;
  }
}

File: backend/crud-application/src/main/java/com/knf/dev/demo/crudapplication/entity/Valuation.java
package com.knf.dev.demo.crudapplication.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "valuations")
@JsonIgnoreProperties("child")
public class Valuation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ElementCollection
    @Column(name = "value")
    private List<String> values;

    @Column(name = "child_id")
    private Long childId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "child_id", referencedColumnName = "id", insertable = false, updatable = false)
    private Child child;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public List<String> getValues() {
        return values;
    }

    public void setValues(List<String> values) {
        this.values = values;
    }

    public Long getChildId() {
        return childId;
    }

    public void setChildId(Long childId) {
        this.childId = childId;
    }

    public Child getChild() {
        return child;
    }

    public void setChild(Child child) {
        this.child = child;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    @Override
    public String toString() {
        return "Valuation{" +
                "id=" + id +
                ", values=" + values +
                ", childId=" + childId +
                ", createdAt=" + createdAt +
                '}';
    }
}
